src/components/shapes/attributes.js

```diff
@@ -235,8 +235,30 @@ module.exports = templatedArray('shape', {
         role: 'info',
         editType: 'arraydraw',
         description: [
-            'Sets the color filling the shape\'s interior.'
+            'Sets the color filling the shape\'s interior. Only applies to closed shapes.'
         ].join(' ')
     },
+    fillrule: {
+        valType: 'enumerated',
+        values: ['evenodd', 'nonzero'],
+        dflt: 'evenodd',
+        role: 'info',
+        editType: 'arraydraw',
+        description: [
+            'Determines which regions of complex paths constitute the interior.',
+            'For more info please visit https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/fill-rule'
+        ].join(' ')
+    },
+    editable: {
+        valType: 'boolean',
+        role: 'info',
+        dflt: false,
+        editType: 'calc+arraydraw',
+        description: [
+            'Determines whether the shape could be activated for edit or not.',
+            'Please note that setting to *false* has no effect in case `config.editable` is set to true'
```

Comment:
Is it just `false` that has no effect if `config.editable` is `true` (or more specifically `config.edits.shapePosition`)? I thought if the old editable mode was enabled then for the moment the new one with activation would be available at all. If that's the case, this should say something like:
```
Has no effect when the older editable shapes mode is enabled via
`config.editable` or `config.edits.shapePosition`
```

Reply:
Description is updated by https://github.com/plotly/plotly.js/pull/4775/commits/d99b44b414b8f90feca4f285a579d0cc6306f42f.

And here is the function that should control the two, I think.
https://github.com/plotly/plotly.js/blob/63ef7aaf311858101c7ed3075a40472eb6341cd1/src/components/shapes/draw.js#L74-L77.